### DISCUSSION TITLE
Use Firebase in PushNotifications.swift

### DISCUSSION THREAD

**Discussion by trytuna** [ionic-team/capacitor#3336] [General] [answered]
I am playing around with Firebase PushNotifications and stubled upon the unimplemented `createChannel()` method. I want to address that by using the Firebase framework.

But I am having trouble importing `FirebaseMessaging` in `PushNotifications.swift`. It says `No such module 'FirebaseMessaging'`. Unfortunately I am not an expert of the Xcode build tools. Could someone kindly guide me through that?

**Comment by jcesarmobile** [comment, marked as answer]
You can't modify PushNotifications.swift because that's Capacitor code.

Unless you want to work in your own fork of Capacitor with a Firebase dependency, in that case you can do as this PR did
https://github.com/ionic-team/capacitor/pull/2738

**Reply by trytuna** [reply]
That helps. Thanks!
